pipeline/config.go

```diff
@@ -53,12 +59,27 @@ func (c Config) BuildPipeline(bc operator.BuildContext, defaultOperator operator
 	return NewDirectedPipeline(operators)
 }
 
-func getBuildContextWithDefaultOutput(configs []operator.Config, i int, bc operator.BuildContext) operator.BuildContext {
-	if i+1 >= len(configs) {
-		return bc
+func SetOutputIDs(operators []operator.Operator, pluginMap map[string]string) error {
+	for i, op := range operators {
+		if len(op.GetOutputIDs()) == 0 && i+1 < len(operators) {
```

Comment:
For the sake of cognitive complexity, I think it would be preferable to check the `i+1 < len(operators)` condition right at the beginning of the loop, and to include a brief comment on why:
```go
// The last operator does not have a next operator to output to
if i+1 < len(operators) {
  break
}
```
Or just convert this to `for i := 0; i+1 < len(operators); i++ { ... }`

Reply:
If I understand the rest of the function correctly, the last operator will proceed to eventually have `SetOutputIDs` called on it, which doesn't seem right. Maybe I'm misinterpreting though.

Reply:
Reiterating request to please add a comment explaining why the loop does not run on the last operator

Reply:
because no output is specified at this point for the last operator, it will always be empty and there is nothing after it to automatically point towards.